ecs-cli/modules/config/config_test.go

```diff
@@ -14,86 +14,106 @@
+	if envAwsAccessKey != resolvedCredentials.AccessKeyID {
+		t.Errorf("Invalid access key set. Expected [%s]. Got [%s]", envAwsAccessKey, resolvedCredentials.AccessKeyID)
 	}
-	// This would be the 'default' profile creds provider in this case.
-	if profileName != profileCredsProvider.Profile {
-		t.Errorf("Invalid profile name set. Expected [%s]. Got [%s]", profileName, profileCredsProvider.Profile)
+	if envAwsSecretKey != resolvedCredentials.SecretAccessKey {
+		t.Errorf("Invalid secret key set. Expected [%s]. Got [%s]", envAwsSecretKey, resolvedCredentials.SecretAccessKey)
 	}
 }
 
-func TestGetCredentialProvidersWithCredentials(t *testing.T) {
+func TestToServiceConfigWhenUsingECSProfileCredentials(t *testing.T) {
 	ecsConfig := NewCliConfig(clusterName)
+	ecsConfig.Region = region
+
 	ecsConfig.AwsAccessKey = awsAccessKey
 	ecsConfig.AwsSecretKey = awsSecretKey
 
-	credentialProviders := ecsConfig.getCredentialProviders()
-	if len(credentialProviders) != 4 {
-		t.Fatal("Unexpected number of credential providers in the chain: ", len(credentialProviders))
+	awsConfig, _ := ecsConfig.ToServiceConfig()
+	resolvedCredentials, err := awsConfig.Credentials.Get()
+	if err != nil {
+		t.Error("Error fetching credentials from the chain provider")
 	}
 
-	// credentialProviders is composed of:
-	// 0 -> static creds provider
-	// 1 -> env provider
-	// 2 -> default profile provider
-	// 3 -> role provider
-	staticCredsProvider, ok := credentialProviders[0].(*credentials.StaticProvider)
-	if !ok {
-		t.Fatal("Mismatch in credential provider chain. Expected to use static creds provider")
+	if awsAccessKey != resolvedCredentials.AccessKeyID {
+		t.Errorf("Invalid access key set. Expected [%s]. Got [%s]", awsAccessKey, resolvedCredentials.AccessKeyID)
 	}
-	if awsAccessKey != staticCredsProvider.AccessKeyID {
-		t.Errorf("Invalid access key set. Expected [%s]. Got [%s]", awsAccessKey, staticCredsProvider.AccessKeyID)
+	if awsSecretKey != resolvedCredentials.SecretAccessKey {
+		t.Errorf("Invalid secret key set. Expected [%s]. Got [%s]", awsSecretKey, resolvedCredentials.SecretAccessKey)
 	}
-	if awsSecretKey != staticCredsProvider.SecretAccessKey {
-		t.Errorf("Invalid secret key set. Expected [%s]. Got [%s]", awsSecretKey, staticCredsProvider.SecretAccessKey)
+}
+
+// TODO Add proper tests for the shared credential provider resolution
+func TestToServiceConfigWhenAWSProfileSpecified(t *testing.T) {
+	t.Skip("Implement me")
+}
+func TestToServiceConfigWhenNoAWSProfileSpecified(t *testing.T) {
+	t.Skip("Implement me")
+}
+func TestToServiceConfigWhenUsingEC2InstanceRole(t *testing.T) {
+	t.Skip("Implement me")
+}
+
+func TestToServiceConfigWhenRegionIsNotSpecified(t *testing.T) {
+	ecsConfig := NewCliConfig(clusterName)
+
+	_, err := ecsConfig.ToServiceConfig()
+	if err == nil {
+		t.Error("There should always be an error when region is not specified in the ecsConfig.")
+	}
+}
+func TestToServiceConfigWhenNoCredentialsAreAvailable(t *testing.T) {
```

Comment:
I think we should skip this test until we either mock out the providers or, at least, mock out the ec2 role provider to an extent (e.g. inject a mock http client). Maybe this is a use-case for overriding the provider chain.

I say that because this test will fail on an EC2 instance with a role attached or could take 5 seconds to pass due to the http timeout.